src/main/java/executor/service/config/bean/BeanConfiguration.java

```diff
@@ -0,0 +1,63 @@
+package executor.service.config.bean;
+
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.ProxyCredentials;
+import executor.service.model.ProxyNetworkConfig;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configuration class responsible for defining and configuring Spring beans.
+ * <p>
+ * This class contains methods annotated with {@code @Bean}, which define and configure
+ * beans that can be managed by the Spring IoC container.
+ * <p>
+ * The {@code threadPoolExecutor} bean creates an {@link ExecutorService} based on the
+ * specified properties, including core pool size, maximum pool size, and keep-alive time.
+ * <p>
+ * The {@code proxyConfigHolder} bean creates and configures a {@link ProxyConfigHolder} bean,
+ * which is composed of a {@link ProxyNetworkConfig} and {@link ProxyCredentials}.
+ *
+ * @author Dima Silenko, Oleksandr Tuleninov
+ * @version 01
+ */
+@Configuration
+public class BeanConfiguration {
```

Comment:
it's useless name witch describe anything or too common things, you will collect all bean configurations in this class? You will mix all layer's configurations  in one class? Or you wouldn't divide configurations on task area?
You decided don't use another cases of decomposition's strategies?

Reply:
Separating into separate classes is a good idea. I did it. Commit f05e7d0a.